frontend/src/components/dashboard/varianceHomePage/VarianceHomePage.js

```diff
@@ -145,44 +145,67 @@ export class VarianceHomePage extends Component {
     this.props.fetchVarianceDocumentCategoryOptions();
   }
 
+  componentWillReceiveProps(nextProps) {
+    const locationChanged = nextProps.location !== this.props.location;
+    if (locationChanged) {
+      this.renderDataFromURL(nextProps);
+    }
+  }
+
   componentWillUnmount() {
     this.handleVarianceSearchDebounced.cancel();
     this.setState({
       params: {},
     });
   }
 
-  renderDataFromURL = (params) => {
+  renderDataFromURL = (nextProps) => {
+    const params = nextProps ? nextProps.location.search : this.props.location.search;
     const { region, compliance_code, major, search, ...remainingParams } = queryString.parse(
       params
     );
-    const format = (param) => (param ? param.split(",").filter((x) => x) : []);
-    this.setState({
-      params: {
-        region: format(region),
-        compliance_code: format(compliance_code),
-        major,
-        search,
-        ...remainingParams,
+    this.setState(
+      {
+        params: {
+          region: formatParamStringToArray(region),
+          compliance_code: formatParamStringToArray(compliance_code),
+          major,
+          search,
+          ...remainingParams,
+        },
       },
-    });
+      () => {
+        this.props.fetchVariances(this.state.params);
+      }
+    );
   };
 
   handleVarianceSearch = (searchParams, clear = false) => {
     const formattedSearchParams = formatParams(searchParams);
+    // const formattedSearchParams = clear ? {} : formatParams(searchParams);
+
     const persistedParams = clear ? {} : formatParams(this.state.params);
+    const updatedParams = {
+      // Start from existing state
+      ...persistedParams,
+      // Overwrite prev params with any newly provided search params
+      ...formattedSearchParams,
+      // Reset page number
+      page: String.DEFAULT_PAGE,
+      // Retain per_page if present
+      // eslint-disable-next-line react/no-access-state-in-setstate
+      per_page: this.state.params.per_page ? this.state.params.per_page : String.DEFAULT_PER_PAGE,
+    };
 
-    this.props.history.push(
-      router.VARIANCE_DASHBOARD.dynamicRoute({
-        // Start from existing state
-        ...persistedParams,
-        // Overwrite prev params with any newly provided search params
-        ...formattedSearchParams,
-        // Reset page number
-        page: String.DEFAULT_PAGE,
-        // Retain per_page if present
-        per_page: this.state.params.per_page ? this.state.params.per_page : String.DEFAULT_PER_PAGE,
-      })
+    this.props.history.push(router.VARIANCE_DASHBOARD.dynamicRoute(updatedParams));
+    this.setState(
+      {
+        params: updatedParams,
+      },
+      // Fetch parties once state has been updated
+      () => {
+        return this.renderDataFromURL();
```

Comment:
`this.props.history.push(router.VARIANCE_DASHBOARD.dynamicRoute(updatedParams));`
this will trigger `ComponentWillReceiveProps` to detect a change in the location, which will invoke `renderdataFromUrl`, so the last return is not necessary.

Reply:
I wouldn't have thought of this, implemented.